### PR TITLE
fix: add telemetry events

### PR DESCRIPTION
**Pull request by saravmajestic**
### Problem
Telemetry events are bloated

### Solution
- consolidate the events that could be associated with features, easily searchable
- proper naming conventions
- applied the above for documentation editor events

### Screenshot/Demo

A picture is worth a thousand words. Please highlight the changes if applicable.

### How to test
- Open doc editor
- perform operations like modifying desc, sync with db, bulk generate etc and check telemetry events

## Checklist

- [ ] I have run this code and it appears to resolve the stated issue
- [ ] `README.md` updated and added information about my change

### CODE REVIEW

File: src/manifest/dbtProject.ts [hunk 1]
```diff
@@ -695,21 +696,40 @@ export class DBTProject implements Disposable {
...
-      "finding distinct values for column",
-      true,
-      { model, column },
-    );
-    const query = `select ${column} from {{ ref('${model}')}} group by ${column}`;
-    const queryExecution = await this.dbtProjectIntegration.executeSQL(
-      query,
-      100, // setting this 100 as executeSql needs a limit and distinct values will be usually less in number
-      model,
+    this.telemetry.startTelemetryEvent(
```
Comment (saravmajestic) [inline src/manifest/dbtProject.ts:699]: wrapped code with try catch block and added telemetry events

Comment (saravmajestic) [inline src/manifest/dbtProject.ts:699]: `startTelemetryEvent` will start tracking the time taken for completing an event

File: webview_panels/src/modules/documentationEditor/components/tests/forms/TestForm.tsx [hunk 2]
```diff
@@ -16,6 +16,8 @@ import Relationships from "./Relationships";
 import { useEffect, useMemo } from "react";
 import useTestFormSave, { TestOperation } from "../hooks/useTestFormSave";
 import { SaveRequest } from "../types";
+import { sendTelemetryEvent } from "../../telemetry";
+import { TelemetryEvents } from "../../../../../../../src/telemetry/events";
```
Comment (anandgupta42) [inline webview_panels/src/modules/documentationEditor/components/tests/forms/TestForm.tsx:20]: @saravmajestic is this common to provide such relative paths?

Comment (saravmajestic) [reply webview_panels/src/modules/documentationEditor/components/tests/forms/TestForm.tsx:20]: Actually not. We should set alias and use it. Let me fix this

Comment (saravmajestic) [reply webview_panels/src/modules/documentationEditor/components/tests/forms/TestForm.tsx]: done

File: src/telemetry/index.ts [hunk 3]
```diff
@@ -8,19 +8,54 @@ export class TelemetryService implements vscode.Disposable {
...
+  private getFeatureName(eventName: string) {
+    const [featureName, rest] = eventName.split("/");
+    if (rest) {
+      return { feature: featureName };
+    }
+    return {};
+  }
+
+  sendTelemetryEvent(
+    eventName: string, // TODO: should be TelemetryEvents
```
Comment (mdesmet) [inline src/telemetry/index.ts:51]: There is an open TODO here.

File: src/telemetry/index.ts [hunk 4]
```diff
@@ -40,14 +75,15 @@ export class TelemetryService implements vscode.Disposable {
   }
 
   sendTelemetryError(
-    eventName: string,
+    eventName: string, // TODO: should be TelemetryEvents
```
Comment (mdesmet) [inline src/telemetry/index.ts:78]: Open TODO

File: src/webview_provider/docsEditPanel.ts [hunk 5]
```diff
@@ -583,7 +595,7 @@ export class DocsEditViewPanel implements WebviewViewProvider {
               panel: this._panel,
               message,
               project,
-              isBulkGen: true,
+              isBulkGen: message.isBulkGen, // TODO: check this with vipul/surya
```
Comment (mdesmet) [inline src/webview_provider/docsEditPanel.ts:598]: open TODO

**Review by mdesmet — COMMENTED** [review]
LGTM apart from todo's

File: src/telemetry/index.ts (hunk 3 above)
Comment (saravmajestic) [reply src/telemetry/index.ts:51]: This is to be addressed after we migrate all telemetry events to this new format

File: src/telemetry/index.ts (hunk 4 above)
Comment (saravmajestic) [reply src/telemetry/index.ts:78]: This is to be addressed after we migrate all telemetry events to this new format

**Pull request merged**